src/test/java/org/stellar/sdk/KeyPairTest.java

```diff
@@ -86,4 +90,35 @@ public void testSignWithoutSecret() {
       assertEquals("KeyPair does not contain secret key. Use KeyPair.fromSecretSeed method to create a new KeyPair with a secret key.", e.getMessage());
     }
   }
+
+  @Test
+  public void testSignPayloadSigner() {
+    KeyPair keypair = KeyPair.fromSecretSeed(Util.hexToBytes(SEED));
+    // the hint from this keypair is [254,66,4,55]
+
+    byte[] payload = new byte[]{1,2,3,4,5};
+    DecoratedSignature sig = keypair.signPayloadDecorated(payload);
+    Assert.assertArrayEquals(sig.getHint().getSignatureHint(), new byte[]{
+            (byte)(0xFF & 252),
+            (byte)(0xFF & 65),
+            (byte)(0),
+            (byte)(0xFF & 50)});
+
+  }
+
+  @Test
+  public void testSignPayloadSignerLessThanHint() {
+    KeyPair keypair = KeyPair.fromSecretSeed(Util.hexToBytes(SEED));
+    // the hint from this keypair is [254,66,4,55]
+
+    byte[] payload = new byte[]{1,2,3};
+    DecoratedSignature sig = keypair.signPayloadDecorated(payload);
+    // the hint could only be derived off of 3 bytes from payload
+    Assert.assertArrayEquals(sig.getHint().getSignatureHint(), new byte[]{
+            (byte)(255),
+            (byte)(0xFF & 64),
+            (byte)(0xFF & 7),
+            (byte)(0xFF & 55)});
```

Comment:
❓ What is the 0xFF here for? Isn't 0xff & 64 = 0x64? I think you can simply do:
```suggestion
            (byte)(255),
            (byte)(64),
            (byte)(7),
            (byte)(55)});
```
See this as an example: https://go.dev/play/p/qi-HubnfmPA

Reply:
yes, wasn't necessary, wanted to show the 8 bit unsigned value being placed into each byte, and did used that form for all values even the ones below 128